src/metpy/units.py

```diff
@@ -202,48 +226,140 @@ def _get_changed_version(docstring):
-        # pint directly (e.g. "[mass] / [length] / [time]**2). This is for both efficiency
-        # reasons and to ensure that problems with the decorator are caught at import,
-        # rather than runtime.
-        dims = {name: (orig, units.get_dimensionality(orig.replace('dimensionless', '')))
-                for name, orig in bound_units.arguments.items()}
 
-        defaults = {name: sig.parameters[name].default for name in sig.parameters
-                    if sig.parameters[name].default is not Parameter.empty}
+def process_units(
+    input_dimensionalities,
+    output_dimensionalities,
+    output_to=None,
+    ignore_inputs_for_output=None
+):
+    """Wrap a non-Quantity-using function in base units to fully handle units."""
+    def dec(func):
+        sig, dims, defaults = _check_units_outer_helper(func, **input_dimensionalities)
 
         @functools.wraps(func)
         def wrapper(*args, **kwargs):
-            # Match all passed in value to their proper arguments so we can check units
-            bound_args = sig.bind(*args, **kwargs)
-            bad = list(_check_argument_units(bound_args.arguments, defaults, dims))
-
-            # If there are any bad units, emit a proper error message making it clear
-            # what went wrong.
-            if bad:
-                msg = f'`{func.__name__}` given arguments with incorrect units: '
-                msg += ', '.join(f'`{arg}` requires "{req}" but given "{given}"'
-                                 for arg, given, req in bad)
-                if 'none' in msg:
-                    msg += ('\nAny variable `x` can be assigned a unit as follows:\n'
-                            '    from metpy.units import units\n'
-                            '    x = units.Quantity(x, "m/s")')
-
-                # If function has changed, mention that fact
-                if func.__doc__:
-                    changed_version = _get_changed_version(func.__doc__)
-                    if changed_version:
-                        msg = (f'This function changed in {changed_version}--double check '
-                               'that the function is being called properly.\n') + msg
-                raise ValueError(msg)
-            return func(*args, **kwargs)
+            bound_args = _check_units_inner_helper(func, sig, defaults, dims, *args, **kwargs)
+
+            # Determine unit(s) with which to wrap output (first, since we mutate the bound
+            # args)
+            if isinstance(output_dimensionalities, tuple):
+                multiple_output = True
+                outputs = output_dimensionalities
+            else:
+                multiple_output = False
+                outputs = (output_dimensionalities,)
+            output_control = []
+            for i, output in enumerate(outputs):
+                convert_to = (
+                    output_to if not multiple_output or output_to is None else output_to[i]
+                )
```

Comment:
Not holding up the PR for this, but this is the kind of garbage that black does that drives me nuts.